### PR TITLE
Docs: update "Apollo Best Coding Practice"

### PR DESCRIPTION
**Pull request by storypku**

### CODE REVIEW

File: docs/technical_tutorial/apollo_best_coding_practice.md [hunk 1]
```diff
@@ -71,14 +72,59 @@
...
+
+   ```
+   cc_library(
+    name = "sandwich",
+    srcs = ["sandwich.cc"],
+    hdrs = ["sandwich.h"],
+    deps = [
+        ":bread",
+        # BAD practice to uncomment the line below
+        # ":floor",
```
Comment (changsh726) [inline docs/technical_tutorial/apollo_best_coding_practice.md:102]: flour

File: docs/technical_tutorial/apollo_best_coding_practice.md [hunk 2]
```diff
@@ -71,14 +72,59 @@
...
+        name = "flour",
+        srcs = ["flour.cc"],
+        hdrs = ["flour.h"],
+    )
+   ```
+
+1. Conform to the DRY principle.
 
-   Don't repeat yourself, in any way. Avoid duplicate classes, functions, const
+   Don't repeat rourself, in any way. Avoid duplicate classes, functions, const
```
Comment (changsh726) [inline docs/technical_tutorial/apollo_best_coding_practice.md:122]: yourself?

File: docs/technical_tutorial/apollo_best_coding_practice.md [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
    foobar.h
    foobar.cc
    foobar_test.cc
    ```
 
-1. A bazel target contains at most one header and one source file.
+1. A bazel target should contain at most one header and one source file.
 
-   ```text
+   ```
```
Comment (changsh726) [inline docs/technical_tutorial/apollo_best_coding_practice.md:19]: Can be marked as python to improve the display.

File: docs/technical_tutorial/apollo_best_coding_practice.md (hunk 1 above)
Comment (storypku) [reply docs/technical_tutorial/apollo_best_coding_practice.md]: Thanks for pointing out!

File: docs/technical_tutorial/apollo_best_coding_practice.md (hunk 3 above)
Comment (storypku) [reply docs/technical_tutorial/apollo_best_coding_practice.md]: Done.

**Review by changsh726 — APPROVED** [review]

**Pull request merged**